projects/dynamicforms/src/components/upload/upload.component.ts

```diff
@@ -116,8 +146,8 @@ export class UploadComponent extends AbstractFormComponent implements OnInit {
         .subscribe()
     );
     this.fileList.splice(index, 1);
-    this.setValueAndValidate(this.fileList);
-
+    this.files.splice(index, 1);
+    this.setValueAndValidate(this.files);
```

Comment:
If it doesn't impact something else, I would leave triggering validation as it was (with `this.fileList`), because it reflects the state on UI and for consistency purposes. Otherwise, this would be the only place in this file where we trigger validation with `this.files`.
Also have you checked if this approach is maybe also needed in `removeAll` method? `this.files = [];` maybe?

Reply:
In removeAll, all documents are removed, html value of input field is ok.
I could  leave this set value and validate for the fileList, but still we need to set value of upload control to the this.files, because in opposite it doesn't work, wrong payload is sent to the backend. 
So in that case we would have weird logic, which first sets value of control to the fileList and immediately after that to the this.files. It's very confusing. Validation with current solution is also working.